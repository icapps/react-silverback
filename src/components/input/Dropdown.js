import React from 'react';
import PropTypes from 'prop-types';

const Dropdown = ({ id, label, value, handleChange, options, isLabelShown }) => (
  <div className="form-group">
    {isLabelShown && <label htmlFor={id}>{label}</label>}
    <select className="form-control" id={id} value={value} onChange={handleChange}>
      {options.map(option => (
        <option key={option.key} value={option.key}>
          {option.text}
        </option>
      ))}
    </select>
  </div>
);

Dropdown.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  handleChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.object])).isRequired,
  isLabelShown: PropTypes.bool,
};

Dropdown.defaultProps = {
  label: '',
  isLabelShown: true,
};

export default Dropdown;
