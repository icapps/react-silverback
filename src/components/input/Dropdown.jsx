import React from 'react';
import PropTypes from 'prop-types';

const Dropdown = props => (
  <div className="form-group">
    <label htmlFor={props.id}>{props.label}</label>
    <select className="form-control" id={props.id} value={props.value} onChange={props.handleChange}>
      {props.options.map(option => <option key={option} value={option}>{option}</option>)}
    </select>
  </div>
);

Dropdown.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Dropdown;
