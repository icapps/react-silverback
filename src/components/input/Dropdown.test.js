import React from 'react';
import { shallow, configure } from 'enzyme';
import Dropdown from './Dropdown';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

describe('Dropdown Component', () => {
  it('should render a Dropdown component', () => {
    const wrapper = shallow(<Dropdown id="test" label="This is a dropdown" value="test" handleChange={() => { }} options={[{ key: 'test', text: 'test' }, { key: 'test2', text: 'test2' }, { key: 'test3', text: 'test3' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
