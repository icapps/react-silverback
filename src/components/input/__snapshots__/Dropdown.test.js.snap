// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component should render a Dropdown component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    handleChange={[Function]}
    id="test"
    isLabelShown={true}
    label="This is a dropdown"
    options={
        Array [
            "test",
            "test2",
            "test3",
          ]
    }
    value="test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test"
>
          This is a dropdown
</label>,
        <select
          className="form-control"
          id="test"
          onChange={[Function]}
          value="test"
>
          <option
                    value={undefined}
          />
          <option
                    value={undefined}
          />
          <option
                    value={undefined}
          />
</select>,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is a dropdown",
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": "This is a dropdown",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value={undefined}
/>,
            <option
              value={undefined}
/>,
            <option
              value={undefined}
/>,
          ],
          "className": "form-control",
          "id": "test",
          "onChange": [Function],
          "value": "test",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test"
>
            This is a dropdown
</label>,
          <select
            className="form-control"
            id="test"
            onChange={[Function]}
            value="test"
>
            <option
                        value={undefined}
            />
            <option
                        value={undefined}
            />
            <option
                        value={undefined}
            />
</select>,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is a dropdown",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "This is a dropdown",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value={undefined}
/>,
              <option
                value={undefined}
/>,
              <option
                value={undefined}
/>,
            ],
            "className": "form-control",
            "id": "test",
            "onChange": [Function],
            "value": "test",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
